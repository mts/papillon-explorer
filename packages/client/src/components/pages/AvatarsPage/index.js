import React from 'react';
import { Link } from 'react-router-dom';
import { startPageRoute } from '../../../routes'

const AvatarsPage = () => {
  return (
    <div>
      <h2>Avatars Page</h2>
      <div>
        Under construction ğŸ™‚
        <Link to={startPageRoute}> Back to Home Page</Link>
      </div>
    </div>
  )
}

export default AvatarsPage;
