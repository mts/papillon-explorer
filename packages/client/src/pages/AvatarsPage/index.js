import React from 'react'
import { Query } from 'react-apollo'
import { Link } from 'react-router-dom'
import { startPageRoute } from '../../routes'
import { PAGES } from '../../graphql/query'

const AvatarsPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? (
          <div>
            <h2>{data.pages.pages.find(page => page.id === 'avatars').name}</h2>
            <div>
              Under construction ğŸ™‚
              <Link to={startPageRoute}> Back to Home Page</Link>
            </div>
          </div>
        ) : null
      }
    </Query>
  )
}

export default AvatarsPage
