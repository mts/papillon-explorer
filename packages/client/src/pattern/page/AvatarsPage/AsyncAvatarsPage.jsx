import React from 'react'

const LazyAvatarsPage = React.lazy(() => import('./AvatarsPage'))

export const AsyncAvatarsPage = () => (
  <React.Suspense fallback={null}>
    <LazyAvatarsPage />
  </React.Suspense>
)
