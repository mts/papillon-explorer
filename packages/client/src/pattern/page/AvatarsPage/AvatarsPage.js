import React from 'react'
import { Query } from 'react-apollo'
import { startPageRoute } from '../../../routes'
import { PAGES } from '../../../graphql/query'
import AvatarsTemplate from '../../template/AvatarsTemplate'

const AvatarsPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? (
          <AvatarsTemplate page={data.pages.pages.find(page => page.id === 'avatars')} startPageRoute={startPageRoute} />
        ) : null
      }
    </Query>
  )
}

export default AvatarsPage
