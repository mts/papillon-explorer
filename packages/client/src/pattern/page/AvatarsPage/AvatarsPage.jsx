import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import { PAGES } from '../../../graphql/query'
import { AvatarsTemplate } from '../../template/AvatarsTemplate'

export const AvatarsPage = () => {
  const { data } = useQuery(PAGES)

  if (data?.pages?.pages) {
    return <AvatarsTemplate page={data.pages.pages.find(page => page.id === 'avatars')} />
  }

  return null
}

export default AvatarsPage
