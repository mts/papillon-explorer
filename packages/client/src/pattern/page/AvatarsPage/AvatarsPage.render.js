import React from 'react'
import { AvatarsPage } from './AvatarsPage'

export const avatarsPageDefault = <AvatarsPage />
