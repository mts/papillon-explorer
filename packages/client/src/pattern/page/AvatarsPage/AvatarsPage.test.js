import React from 'react'
import { avatarsPageDefault } from './AvatarsPage.render'

jest.mock('@apollo/react-hooks', () => {
  return {
    useQuery: () => {
      return {
        data: {
          pages: {
            pages: [{ id: 'avatars', name: 'some-name' }],
          },
        },
      }
    },
  }
})

jest.mock('react-router-dom', () => ({ Link: () => <div>some link</div> }))

describe('<AvatarsPage />', () => {
  describe('Snaphot', () => {
    test('must match avatarsPageDefault', () => {
      expect(global.renderToJSON(avatarsPageDefault)).toMatchSnapshot()
    })
  })
})
