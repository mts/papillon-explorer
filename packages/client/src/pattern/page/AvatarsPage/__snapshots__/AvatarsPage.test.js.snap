// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarsPage /> Snaphot must match avatarsPageAsync 1`] = `null`;

exports[`<AvatarsPage /> Snaphot must match avatarsPageDefault 1`] = `
<div>
  <h2>
    some-name
  </h2>
  <div>
    Under construction
     
    <span
      aria-label="emoji"
      role="img"
    >
      ðŸ™‚
    </span>
    <div>
      some link
    </div>
  </div>
</div>
`;

exports[`<AvatarsPage /> Snaphot must match avatarsPageNull 1`] = `null`;
